src/test/datascience/jupyter/kernels/installationPrompts.vscode.test.ts

```diff
@@ -272,20 +292,201 @@ suite('DataScience Install IPyKernel (slow) (install)', function () {
 
         // The prompt should be displayed when we run a cell.
         await waitForCondition(async () => prompt.displayed.then(() => true), delayForUITest, 'Prompt not displayed');
+        await verifyIPyKernelPromptDisplayed(prompt, venvPythonPath);
+        await verifyErrorInCellOutput(notebookDocument, venvPythonPath);
+
+        // Submitting code again should display the same prompt again.
+        prompt.reset();
+        await activeInteractiveWindow.addCode(source, untitledPythonFile.uri, 0).catch(noop);
+        await verifyIPyKernelPromptDisplayed(prompt, venvPythonPath);
+        await verifyErrorInCellOutput(notebookDocument, venvPythonPath);
+
+        // Submitting code again should display the same prompt again.
+        prompt.reset();
+        await activeInteractiveWindow.addCode(source, untitledPythonFile.uri, 0).catch(noop);
+        await verifyIPyKernelPromptDisplayed(prompt, venvPythonPath);
+        await verifyErrorInCellOutput(notebookDocument, venvPythonPath);
+
+        await sleep(1_000);
+
+        // Verify we didn't get a prompt again.
+        // In the past when we dismissed the prompt, we would get a prompt again.
+        assert.strictEqual(prompt.getDisplayCount(), 1, 'Should not display additional prompts');
+
+        prompt.dispose();
+        // Next, lets try to run again, but select a kernel that doesn't have ipykernel.
+        // At this point we should get the prompt again for this new kernel.
+        const promptOptions: WindowPromptStubButtonClickOptions = {
+            text: DataScience.selectKernel(),
+            clickImmediately: true
+        };
+        prompt = await hijackPrompt(
+            'showInformationMessage',
+            { contains: expectedPromptMessageSuffix },
+            promptOptions,
+            disposables
+        );
 
-        const cell = notebookDocument.cellAt(0)!;
-        assert.equal(cell.outputs.length, 0);
+        // Hijack the select kernel functionality so it selects the correct kernel
+        const stub = sinon.stub(kernelSelector, 'selectKernel').callsFake(async function () {
+            // Now that we have selected a kernel, next time we get the prompt again, just dismiss the prompt.
+            promptOptions.dismissPrompt = true;
+            delete promptOptions.text;
+            // In tests, things hang as the IW isn't focused.
+            void activeInteractiveWindow.show(false);
```

Comment:
This is a very very weird issue.
The tests fail locally because the IW is no longer the active editor.
Hence when we get the kernel selection code, it fails as that code checks whether there's an active notebook editor, and since its null, that method waits indefinitely for something to get active.